### ISSUE TITLE
Request for Comments: Future of @ng-idle

### ISSUE THREAD

**Issue by HackedByChinese** [HackedByChinese/ng2-idle#97]
Greetings, all!

I know a lot of you are wondering about the status of this project and are perhaps (and understandably) annoyed about turn around time and status of issues.

First, I still feel the use cases and implementation pattern of this module (and ng-idle) are distinct and useful from other solutions out there. However, I have an admission to make: I've been cheating on you (the Angular community). Most of my work in the past year and a half has been with React. This means that I'm just not invested into Angular like I might have been in the past, which I suppose makes me a bit of a liability to the project. So, in the short term, it's a good idea for the project to be handled over to a new lead contributor that can keep it maintained and grow it alongside Angular.

Second, I mentioned my interest in the architecture. So much so that I've replicated it a few times now (original was jQuery, then AngularJS, and then Angular... I've even got a stripped down version for React). I'm still interested in it! However, I'm thinking that maybe we can create a new org (IdleJS or something like that) where the core functionality is defined as vanilla JS, and we have framework-specific adapter modules for things like Angular and React. That way the entire community can enjoy a consistent implementation pattern for these use cases, and fans of a particular framework can concentrate on adapting it for their environment. I suppose the biggest hitch would be whether or not we can make a consistent core that works for popular frameworks that isn't encumbered, dependency-heavy, or too large.

Anyways, I'm interested to hear what you think. Also, those that are interested in helming the current project, definitely speak up.

**Comment by plisovyi** [comment]
Hey. Although IdleJS is a great idea (not kidding, having ideology and architecture of solution and multiple implementations are just great in my books) I do think you should handle over the control if you lack the time or desire to continue. I saw there's PR with updated required for use with version 6 of RxJS and Angular and there's no response for some time already. 
So, my vote is — get org going, while handling keys to someone willing to too.

**Comment by HackedByChinese** [comment]
@plisovyi Yes, that was my intention was regardless, hand over ng-idle as-is and work on a cross platform version later.

**Comment by MHacker9404** [comment]
I'd be more than willing to help, not necessarily as maintainer, but I'd love to contribute as best I can

**Comment by moribvndvs** [comment]
Hey, all. I just wanted to bump this up again to get visibility. I had been working with a group to take over maintenance but I think we both fizzled out and the main contact I had left that original group.

**Comment by bh3605** [comment]
I think this project is just small enough that I think I could put time in maintaining it. The project as is, I wouldn't know what would need updating until I scoured the entire codebase and started fixing some of the reported bugs on it. I do have experience updating npm packages at my current organization.

**Comment by moribvndvs** [comment]
Thanks for stepping up! Fortunately i think version updates aren’t too bad. I believe the CLI has a task to handle deps for you. The rest is just dealing with breaking changes and releasing. We should also consider Greenkeeper or something to automate this and relieve some of the burden.

We can discuss in an email or chat on how to proceed if you’re still interested.

**Comment by bh3605** [comment]
Greenkeeper is losing all support. It's moving its current userbase over to https://snyk.io/. Would rather using that over something that's no longer accepting no users haha.

**Comment by moribvndvs** [comment]
Whoops, well fair enough. Well anyway, let me do a little housekeeping here, then I think perhaps we can work out a targeted list to attack with you added as a maintainer, then figure out what to do what you’ve got a handle on it.

**Comment by bh3605** [comment]
Sounds good! Looking forward to this (=

I'm excited to be helping the open source community.

**Comment by worksoncloud** [comment]
@moribvndvs I'm interested in helping to maintain this library as well. Currently working with a client who has a need for this library in an Angular 11 SSR scenario. Beyond that, I have been looking for an open source project to contribute to in my spare time and I have ~5 years of real world Angular experience to bring to the table.

**Comment by SuperITMan** [comment]
@moribvndvs I'm interested too for helping to maintain the library.
We use your library in our framework [NationalBankBelgium/stark](https://github.com/NationalBankBelgium/stark) and we're currently updating it to Angular 12 😊 
I have ~5 years of Angular experience.

**Comment by moribvndvs** [comment]
Hey sorry about my negligence to the interested parties. I'm really glad to hear about the interest in maintaining the project. Full disclosure: I've never handed off maintenance in an OSS project before, so bear with me.

Here are some general thoughts I have:

* I would prefer not to just transfer the whole repo to someone else. My concern there is purely that if that person no longer wants to work on the project, it'll be hard to get it back and the project could get abandoned. My preference is to keep on as an owner in its current place and provide guidance on the history and spirit of the project, but get out of the way of maintainers who would otherwise operate self sufficiently. I am open to the idea of moving it to an ng-idle organization as a neutral ground, as long as I can still be an owner and not get kicked out (again, purely just so the project doesn't get abandoned and lost).
* Multiple maintainers sounds like a good idea at this point, to share the load. I think we probably still need a lead maintainer. I'm not sure how to go about this in a fair way other than asking for people to a) state whether or not they're interested in being lead maintainer b) pick or elect from those still interested based on 1. OSS project maintenance experience 2. ability to commit to the role 3. Angular experience 4. ng-idle project experience. It maybe feels like I'm overthinking it, but I do feel it's a good idea to make a solid choice in this matter before I hand over keys to publish packages. Thoughts?
* Once I can get a feel for who's the lead vs who's a maintainer, here are the things I think need to get done:

1. Ensure lead can manage repo settings, CI/coverage, and publish to npm
2. Get releases for the newer versions of Angular out
3. Fix high priority bugs
4. Update demos (move to in-repo so they're up to date and easier to maintain) and documentation
5. Get snyk or whatever working to automated updating deps and submitting PRs to help alleviate the burden

**Comment by SuperITMan** [comment]
Hello,

I think it's a good idea that you keep the ownership on the project for multiple reasons. This can also be done through an "ng-idle" organization 😊 

For my part, I can help to implement CI/CD with GitHub Actions, coveralls, dependabot, snyk, SonarCloud,... as I have some experience with them. I'll do a first implementation and share it through a PR this week or the week after.
For the rest, I can help to update the project with Angular updates with other people 👍

**Comment by bh3605** [comment]
I'm interested in being the lead maintainer. I maintain a couple of my employer's own private npm packages so addressing high priority bugs, documenting everything, and publishing to npm is peanuts to me. =)

**Comment by SuperITMan** [comment]
Hey @bh3605 @worksoncloud @moribvndvs 

As promised last week, I created a 1st PR to switch from Travis CI to GitHub Actions. Could you have look? 😊

**Comment by moribvndvs** [comment]
@worksoncloud I've kicked this off by adding you as a collaborator. I'm not ignoring everyone else that volunteered, I just don't want to throw everyone at it at once and create chaos.

I did cover above the first things I think need to be tackled to get this moving, copied again for your convenience:

1.     Ensure lead can manage repo settings, CI/coverage, and publish to npm
2.     Get releases for the newer versions of Angular out
3.     Fix high priority bugs
4.     Update demos (move to in-repo so they're up to date and easier to maintain) and documentation
5.     Get snyk or whatever working to automated updating deps and submitting PRs to help alleviate the burden

I created a Discord server to facilitate discussions needed to get these things done, hopefully that's acceptable. Just join and shout out in #general if you were one of the collaborator volunteers and I'll add you to the collaborators channel.

https://discord.gg/6ndEd7tf

**Comment by moribvndvs** [comment]
Thanks, @SuperITMan for the GH Actions, nice to have that taken care of.

**Comment by SuperITMan** [comment]
@moribvndvs @worksoncloud

Could you add me as collaborator too? 😊
Like this, I could configure correctly GitHub Actions, Snyk, Dependabot...

**Comment by ryltar** [comment]
@moribvndvs Hi, did you move forward with idleJS ? Looks like to be a great idea btw.

**Comment by seahsky** [comment]
Currently using this in my company application and found this project interesting, can I become a maintainer?

**Comment by benceHornyak** [comment]
Hi @moribvndvs 👋

I hope you are doing great! If you are still looking for new maintainers i'd like to become one. Just wanted to first ask for guidence about what to do around the project. Tried to join the discord server but link was outdated I guess :) 

I've been working with Angular 2+ since 2016 and in the last 1-2 months I've been looking into Angular projects to contribute to and give back to the community somehow  more actively, but I've failed to find good active projects and I'm not confident enough (yet) to build something from scratch 😀 

If you'd be okay to move forward and guide me on what to the next I could work on things at least 5-6hrs per week

**Comment by Eldorados** [comment]
Wondering what the status is on this, has new maintainers been found?

There's a couple PR's open that would be very nice to fix, mainly one that addresses when browsers/tabs are put to 'sleep' and has been open for 4 years now

Also support for Angular 16

**Comment by AdrianRomanski** [comment]
@moribvndvs help is still needed?

**Comment by kikiminyes** [comment]
I'm interested in helping maintain this project.

I started with a low-risk triage pass:
- cloned current `master`
- installed dependencies with `npm ci`
- built both libraries with `npm run build:libs`
- checked #224 against the current `projects/core` and `projects/keepalive` sources and left a reproducibility-focused triage comment

I'm comfortable helping with issue triage, PR review, Angular compatibility fixes, dependency cleanup, tests, and small releases. Happy to begin with limited permissions and low-risk maintenance tasks so trust can be built incrementally.
